components/experimental/src/measure/single_unit_vec.rs

```diff
@@ -28,6 +28,21 @@ impl SingleUnitVec {
             SingleUnitVec::Multi(units) => &units[..],
         }
     }
+
+    pub(crate) fn push(&mut self, unit: SingleUnit) {
+        match self {
+            SingleUnitVec::Zero => *self = SingleUnitVec::One(unit),
+            SingleUnitVec::One(unit) => *self = SingleUnitVec::Two([*unit, *unit]),
```

Comment:
you're duplicating the first value because you're shadowing the argument. this is probably why there's a data diff